### DISCUSSION TITLE
입점되지 않은 대행사의 커머스 API 개발

### DISCUSSION THREAD

**Discussion by dgv7** [commerce-api-naver/commerce-solutions#97] [묻고 답하기] [answered]
개인사업자를 운영중이고, 커머스API를 활용한 서비스를 개발하려는 단계입니다. 스마트 스토어에서 고객과 API 연결 시 입점된 리스트만 노출되고 있는 것을 확인했습니다. 입점된 마케팅 대행사만 고객의 API를 활용할 수 있는 것이 원칙인 것 같은데 입점하지 않은 대행사들은 어떤 방식으로 API를 이용할수 있을까요? 1) 인터넷에 돌아다니는 방법 중에 스마트스토어를 운영중인 고객에게 커머스API 앱을 만드는 것을 유도해서 토큰을 공유받는 방법으로 진행하는 것을 확인했는데 그러한 방식이 유효한 방식일까요? 다른 대안의 방법이 있나요?

주석 1) https://youtu.be/4NGham2hahk?si=8OKvvIPVPD_eAofw

**Comment by commerce-api-naver** [comment, marked as answer]
안녕하세요. 커머스API 이민석입니다.

먼저 말씀해주신 것과 같은 방식(판매자 분들께 커머스API 애플리케이션(내 스토어 애플리케이션)을 유도)은 관련 이용약관 위배의 소지가 있으며 권장하지 않습니다.
뿐만 아니라 판매자 분들의 스마트스토어 보안 문제와도 연결되기에 다양한 채널에서 판매자 분들께는 그러한 방식으로의 이용을 자제할 것을 알려드리고 있습니다.

사용자 분께서 말씀해주신 것과 같이 직접 스마트스토어를 운용하는 것이 아니라 판매자 분들의 스마트스토어 관리 니즈를 대행하는 성격이실 경우 네이버에서 정식으로 제공하는 2가지 수단 중 1개를 택하시는 것을 권해드립니다.
1. 커머스솔루션마켓 입점 (커머스솔루션)
2. API대행사 연동

위 2가지 방법은 솔루션 구독/API대행사 설정과 같이 안전하고 간편한 수단으로 판매자와 서드파티 솔루션 시스템을 연결합니다.
커머스솔루션마켓 입점이 가능한 형태의 서비스를 제공하시는 경우 커머스솔루션마켓 입점을 권장드립니다. 커머스솔루션마켓 입점에 적합하지 않은 서비스를 제공하신다면 API대행사 방식으로 연동해보시기 바랍니다.

- 커머스솔루션마켓 입점 권장 유형: [[커머스API센터 > 솔루션 개발 가이드 > 시작하기 > 마켓 입점 가능 솔루션](https://apicenter.commerce.naver.com/ko/basic/solution-doc#section/%EC%8B%9C%EC%9E%91%ED%95%98%EA%B8%B0/%EB%A7%88%EC%BC%93-%EC%9E%85%EC%A0%90-%EA%B0%80%EB%8A%A5-%EC%86%94%EB%A3%A8%EC%85%98)]
    - 커머스솔루션마켓 입점 과정은 위 개발 가이드 내용을 참고해주세요 (시작하기 > 마켓 입점 프로세스)
- API대행사 연동 공지사항: https://notice.naver.com/notices/cac/14592
